tests/functional/adapter/list_relations_tests/test_special_characters.py

```diff
@@ -0,0 +1,25 @@
+import pytest
+from dbt.tests.util import run_dbt
+
+
+TABLE_BASE_SQL = """
+-- models/my_model.sql
+{{ config(schema = '1_contains_special*character$') }}
+select 1 as id
+"""
+
+
+class TestSpecialCharactersInSchema:
+
+    @pytest.fixture(scope="class")
+    def project_config_update(self):
+        return {"quoting": {"schema": True, "identifier": True}}
```

Comment:
Per a private discussion, I agree that we should not need to specify `identifier` here. I think what's happening is that you're naming the table `table` (from `"table.sql"` below) and that's a reserved word. Try something like `my_table` and see if you can remove `identifier` here.

Reply:
updated to the generic my_model format